### PR TITLE
[Feature] Add compose for iOS

### PR DESCRIPTION
**Pull request by davthecodercom**
+ demo ui

### :tophat: How was this resolved?
- as previously discussed we will develop the KMP version of the app with Compose MP.
- Add compose for ios adapting to current dependency versions

Screenshot 2023-10-03 at 00 24 36

### CODE REVIEW

**Comment by davthecodercom** [comment]
I think the variable missing `CLARITY_PROJECT_ID` is something expected?

```
> Task :gabimoreno:kaptDebugKotlin FAILED
/home/runner/work/Base/Base/gabimoreno/build/generated/source/buildConfig/debug/soy/gabimoreno/BuildConfig.java:12: error: illegal start of expression
  public static final String CLARITY_PROJECT_ID = ;
                                                  ^

> Task :bike:testDebugUnitTest
```

**Comment by soygabimoreno** [comment]
> I think the variable missing `CLARITY_PROJECT_ID` is something expected?
> 
> ```
> > Task :gabimoreno:kaptDebugKotlin FAILED
> /home/runner/work/Base/Base/gabimoreno/build/generated/source/buildConfig/debug/soy/gabimoreno/BuildConfig.java:12: error: illegal start of expression
>   public static final String CLARITY_PROJECT_ID = ;
>                                                   ^
> 
> > Task :bike:testDebugUnitTest
> ```

Do you mean [this](https://github.com/soygabimoreno/Base/pull/28)?

**Comment by davthecodercom** [comment]
> > I think the variable missing `CLARITY_PROJECT_ID` is something expected?
> > ```
> > > Task :gabimoreno:kaptDebugKotlin FAILED
> > /home/runner/work/Base/Base/gabimoreno/build/generated/source/buildConfig/debug/soy/gabimoreno/BuildConfig.java:12: error: illegal start of expression
> >   public static final String CLARITY_PROJECT_ID = ;
> >                                                   ^
> > 
> > > Task :bike:testDebugUnitTest
> > ```
> 
> Do you mean [this](https://github.com/soygabimoreno/Base#configuration)?

I saw it failed in the CI for the variable and I think that config is for my local. 
Is this error message expected?

File: iosApp/iosApp/ComposeView.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+//  ComposeView.swift
+//  iosApp
+//
+//  Created by David Cruz Anaya on 03/10/2023.
+//  Copyright © 2023 orgName. All rights reserved.
+//
+
```
Comment (soygabimoreno) [inline iosApp/iosApp/ComposeView.swift:8]: Surplus code

File: gradle.properties [hunk 2]
```diff
@@ -7,4 +7,14 @@ org.gradle.configuration.cache=true
 org.gradle.daemon=true
 org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
 org.gradle.parallel=true
+
+#MPP
```
Comment (soygabimoreno) [inline gradle.properties:11]: Surplus code

File: iosApp/iosApp/ComposeView.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+struct ComposeView: UIViewControllerRepresentable {
+    func updateUIViewController(_ uiViewController: UIViewControllerType, context: Context) {
+        // no-op
+    }
+    
+    func makeUIViewController(context: Context) -> some UIViewController {
+        AppKt.MainViewController()
+    }
+}
+
```
Comment (soygabimoreno) [inline iosApp/iosApp/ComposeView.swift:22]: 2 empty lines maybe?

File: shared/src/commonMain/kotlin/soy/gabimoreno/shared/App.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    ) {
+        Button(
+            onClick = {
+                count++
+            }
+        ) {
+            Text(text = "Click me: $count")
+        }
+    }
+
```
Comment (soygabimoreno) [inline shared/src/commonMain/kotlin/soy/gabimoreno/shared/App.kt:33]: Surplus empty line

**Review by soygabimoreno — APPROVED** [review]

File: shared/src/commonMain/kotlin/soy/gabimoreno/shared/App.kt (hunk 4 above)
Comment (davthecodercom) [reply shared/src/commonMain/kotlin/soy/gabimoreno/shared/App.kt:33]: done

File: iosApp/iosApp/ComposeView.swift (hunk 3 above)
Comment (davthecodercom) [reply iosApp/iosApp/ComposeView.swift:22]: done

File: iosApp/iosApp/ComposeView.swift (hunk 1 above)
Comment (davthecodercom) [reply iosApp/iosApp/ComposeView.swift:8]: done

File: gradle.properties (hunk 2 above)
Comment (davthecodercom) [reply gradle.properties:11]: done

**Review by soygabimoreno — APPROVED** [review]

**Pull request merged**